src/main/scala/com/databricks/spark/xml/parsers/dom/DomXmlParser.scala

```diff
@@ -148,23 +154,32 @@ private[xml] object DomXmlParser {
 
   def apply(xml: RDD[String],
             schema: StructType,
-            parseMode: String,
-            excludeAttributeFlag: Boolean,
-            treatEmptyValuesAsNulls: Boolean): RDD[Row] = {
+            conf: DomConfiguration): RDD[Row] = {
+    val failFast = conf.failFastFlag
+
     xml.mapPartitions { iter =>
       iter.flatMap { xml =>
-        val builder = DocumentBuilderFactory.newInstance().newDocumentBuilder()
-
         // It does not have to skip for white space, since [[XmlInputFormat]]
         // always finds the root tag without a heading space.
-        val childNode = builder.parse(new ByteArrayInputStream(xml.getBytes))
-          .getChildNodes.item(0)
-        val conf = DomConfiguration(excludeAttributeFlag, treatEmptyValuesAsNulls)
-        val parser = new DomXmlParser(childNode, conf)
-        if (parser.isEmpty) {
-          None
-        } else {
+        val builder = DocumentBuilderFactory.newInstance().newDocumentBuilder()
+
+        // We treats the exception explicitly below. So turn off the error handler.
+        builder.setErrorHandler(null)
+        try{
+          val childNode = builder.parse(new ByteArrayInputStream(xml.getBytes))
+            .getChildNodes.item(0)
+          val parser = new DomXmlParser(childNode, conf)
+          if (parser.isEmpty) {
+            None
+          }
           Some(convertObject(parser, schema))
+        } catch {
+          // Java library DOMParser throws SAXException when it fails.
+          case se: SAXException if !failFast =>
+            logger.warn(s"Dropping malformed row: ${xml.replaceAll("\n", "")}")
+            None
+          case se: SAXException if failFast =>
+            throw new RuntimeException(s"Malformed row (failing fast): ${xml.replaceAll("\n", "")}")
```

Comment:
I wonder if this  flatten xml `${xml.replaceAll("\n", "")}` would be a correct message form or not.